test/m1_facilitator/services/ProveGateway/update.test.ts

```diff
@@ -0,0 +1,163 @@
+// Copyright 2020 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Web3 from 'web3';
+import sinon from 'sinon';
+
+import { ProofGenerator } from '@openst/mosaic-proof';
+import Mosaic from 'Mosaic';
+import BigNumber from 'bignumber.js';
+import * as web3Utils from 'web3-utils';
+
+import Repositories
+  from '../../../../src/m1_facilitator/repositories/Repositories';
+import ProveGatewayService
+  from '../../../../src/m1_facilitator/services/ProveGatewayService';
+import TransactionExecutor
+  from '../../../../src/m1_facilitator/lib/TransactionExecutor';
+import Anchor from '../../../../src/m1_facilitator/models/Anchor';
+import Gateway, { GatewayType } from '../../../../src/m1_facilitator/models/Gateway';
+import Message, {
+  MessageStatus,
+  MessageType,
+} from '../../../../src/m1_facilitator/models/Message';
+import SpyAssert from '../../../test_utils/SpyAssert';
+
+describe('ProveGateway::update', () => {
+  let originTransactionExecutor: TransactionExecutor;
+  let auxiliaryTransactionExecutor: TransactionExecutor;
+  let proveGatewayService: ProveGatewayService;
+  let outboxProofSpy: any;
+  let getERC20CogatewaySpy: any;
+  let proveGatewaySpy: any;
```

Comment:
Disable linter for L41,42,43
`// eslint-disable-next-line @typescript-eslint/no-explicit-any`

Reply:
Done